pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -376,7 +377,7 @@ func (adc *attachDetachController) populateActualStateOfWorld() error {
 			// volume spec is not needed to detach a volume. If the volume is used by a pod, it
 			// its spec can be: this would happen during in the populateDesiredStateOfWorld which
 			// scans the pods and updates their volumes in the ActualStateOfWorld too.
-			err = adc.actualStateOfWorld.MarkVolumeAsAttached(uniqueName, nil /* VolumeSpec */, nodeName, attachedVolume.DevicePath)
+			err = adc.actualStateOfWorld.MarkVolumeAsAttached(uniqueName, nil /* VolumeSpec */, nodeName, attachedVolume.DevicePath, resource.Quantity{Format: resource.BinarySI})
```

Comment:
Something seems wrong here since you are having to pass new params to A/D controller but don't use those new params in A/D controller.

Reply:
yeah I thought about it, it is somewhat ugly. we don't track `pvc.status.capacity` in control-plane for volume expansion in attach-detach controller but we do in volumemanager and hence this parameter is unused in A/D controller.

I could introduce a new interface function but it has to be part of `ActualXXUpdater` interface and A/D controller side of `ASOW` has to provide a dummy implementation.  I am happy to do that instead of this.

Reply:
I have a commit that introduces the new interface function called `MarkAsAttachedWithSize` inside `ActualStateOfWorldAttacherUpdater` interface.  I have not pushed it because I am afraid to make CI unhappy at last minute - https://github.com/gnufied/kubernetes/commit/983e410660fcec7bb0fb019f9d68eb4d7eaf333a#diff-24ba5bb87df94b8b68b6fbd13ef972b0R262 

But I am happy to do this in a follow up. The change is mostly cosmetic and does not changes underlying logic of the PR.

Reply:
done